drivers/adc/adc_stm32.c

```diff
@@ -315,13 +286,13 @@ static int check_buffer(const struct adc_sequence *sequence,
 		return -ENOMEM;
 	}
 
-#if defined(CONFIG_ADC_STM32_DMA) && defined(CONFIG_SOC_SERIES_STM32H7X)
+#if defined(CONFIG_ADC_STM32_DMA)
```

Comment:
```suggestion
#if defined(CONFIG_ADC_STM32_DMA) && defined(CONFIG_SOC_SERIES_STM32H7X)
```

After running tests on this PR in CI for multiple series, there is a regression in the `adc_api` driver test for the `Nucleo_F746ZG` with the following log:
```
Running TESTSUITE adc_basic
===================================================================
START - test_adc_asynchronous_call
E: Supplied buffer is not in a non-cacheable region according to DTS.

    Assertion failed at WEST_TOPDIR/zephyr/tests/drivers/adc/adc_api/src/test_adc.c:223: test_task_asynchronous_call: (ret not equal to 0)
adc_read_async() failed with code -22
 FAIL - test_adc_asynchronous_call in 0.022 seconds
===================================================================
[...]
```

After reintroducing the condition `&& defined(CONFIG_SOC_SERIES_STM32H7X)`, all tests passed for all series. There may be another way to solve the issue, but adding this condition makes everything work well.

Reply:
Any consensus on this? It's a trivial addition and a similar fix seems to be in place for `nucleo_h743zi`.